File: tests/fixtures/swift-simple/Sources/Controllers/OrderController.swift
/// Handles order operations including payment processing.
class OrderController {
    private let paymentService: PaymentServiceProtocol
    private let logger: Logger

    init(paymentService: PaymentServiceProtocol, logger: Logger) {
        self.paymentService = paymentService
        self.logger = logger
    }

    func createOrder(orderId: String, amount: Double) -> Bool {
        logger.info("Creating order: \(orderId)")
        let result = paymentService.processPayment(orderId: orderId, amount: amount)
        if result {
            logger.info("Order created successfully")
        }
        return result
    }

    func cancelOrder(orderId: String) {
        logger.info("Cancelling order: \(orderId)")
        paymentService.refundPayment(transactionId: orderId)
    }

    static func create() -> OrderController {
        let logger = Logger(context: "OrderController")
        let service = PaymentService(logger: logger)
        return OrderController(paymentService: service, logger: logger)
    }
}

File: tests/fixtures/lua-simple/src/controllers/order_controller.lua
-- Handles order operations including payment processing.
local PaymentService = require("payments.payment_service")
local Logger = require("utils.logger")

local OrderController = {}
OrderController.__index = OrderController

function OrderController.new(paymentService, logger)
    local self = setmetatable({}, OrderController)
    self.paymentService = paymentService
    self.logger = logger
    return self
end

function OrderController:createOrder(orderId, amount)
    self.logger:info("Creating order: " .. orderId)
    local result = self.paymentService:processPayment(orderId, amount)
    if result then
        self.logger:info("Order created successfully")
    end
    return result
end

function OrderController:cancelOrder(orderId)
    self.logger:info("Cancelling order: " .. orderId)
    self.paymentService:refundPayment(orderId)
end

return OrderController
